internal/blockchain/utxoviewpoint.go

```diff
@@ -351,27 +358,42 @@ func (view *UtxoViewpoint) connectRegularTransaction(tx *dcrutil.Tx,
 		// details have been accessed since spending it might clear the fields
 		// from memory in the future.
 		entry.Spend()
+
+		// Mark txouts that are spent by transaction inputs later in the regular
+		// transaction tree of the same block as zero conf spends.
+		if inFlightIdx, ok := inFlightTx[prevOut.Hash]; ok && blockIndex >
+			inFlightIdx {
+
+			entry.state |= utxoStateSpentByZeroConf
+		}
 	}
 
 	// Add the transaction's outputs as available utxos.
 	view.AddTxOuts(tx, blockHeight, blockIndex, isTreasuryEnabled)
 
+	// Keep track of in-flight transactions in order detect spends of earlier
+	// outputs by transactions later in the same block.
+	inFlightTx[*tx.Hash()] = blockIndex
 	return nil
 }
 
 // connectRegularTransactions updates the view by adding all new utxos created
 // by the transactions in the regular transaction tree of the block and marking
-// all utxos those same transactions spend as spent.  In addition, when the
-// 'stxos' argument is not nil, it will be updated to append an entry for each
-// spent txout.  An error will be returned if the view does not contain the
-// required utxos.
+// all utxos those same transactions spend as spent.  It also marks all earlier
```

Comment:
s / `...and marking all utxos those same transactions spend as spent.` /  `...and marking all utxos with the same transactions spend as spent.`

Reply:
I don't think that would be correct.  Spend is the verb while spent is the noun.  So, the existing "marking all utxos those same transaction spend (the verb) as spent (the noun)" seems accurate to me.

That said, I'm not opposed to coming up from different wording if you find the way it is currently written to be hard to parse.

Reply:
You're right, it's clearer after a second read. Disregard.